src/core/qgsogrutils.cpp

```diff
@@ -201,7 +202,13 @@ bool QgsOgrUtils::readOgrFeatureGeometry( OGRFeatureH ogrFet, QgsFeature &featur
 {
   if ( !ogrFet )
     return false;
-
+  // Note: this will not work correctly with features that contain more than 1 geometry (GML)
```

Comment:
To make sure, I did another grep on the whole project
* and found that is being used in the tests and in
* QgsOgrUtils::readOgrFeature
   * which also only used in the tests

I think that Nyall Dawson look into this and decide.

`
tests/src/core/testqgsogrutils.cpp:    void readOgrFeatureGeometry();
tests/src/core/testqgsogrutils.cpp:void TestQgsOgrUtils::readOgrFeatureGeometry()
tests/src/core/testqgsogrutils.cpp:  QgsOgrUtils::readOgrFeatureGeometry( nullptr, f );
tests/src/core/testqgsogrutils.cpp:  QgsOgrUtils::readOgrFeatureGeometry( oFeat, f );
`